power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -216,9 +231,14 @@ template <dataset_type_tag dataset_type_> class Dataset {
                 throw DatasetError{"For a non-uniform buffer, indptr should be supplied!\n"};
             }
             if constexpr (std::same_as<check_indptr_content, immutable_t>) {
-                if (indptr[0] != 0 || indptr[batch_size()] != total_elements) {
-                    throw DatasetError{
-                        "For a non-uniform buffer, indptr should begin with 0 and end with total_elements!\n"};
+                auto indptr_span = get_indptr_span(indptr);
+                if (auto decreasing_found = std::ranges::adjacent_find(indptr_span, std::greater<>());
+                    decreasing_found != indptr_span.end()) {
+                    throw DatasetError{"For a non-uniform buffer, indptr must be increasing!\n"};
+                }
+                if (indptr_span.front() < 0 || indptr_span.back() > total_elements) {
```

Comment:
I am against this. Just keep the original requirement: `indptr` begins with `zero` and ends with `total_elements`. That's the standard for all the `indptr` places.

Reply:
then we need to modify the documentation in https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.data_types.SparseBatchArray , which says

> The elements are the indices in the data that point to the first element of that scenario.
> The last element is one after the data index of the last element of the last scenario.
> Usually, the last element will therefore be the size of the data.

Reply:
in this particular case, I would actually even go as far as to make this UB (which just so happens to work as expected) if we feel like explicitly adding this support is going too far

Reply:
Modify the documentation is fine.



> it also would prevent us from making subranges

The user of `Dataset` can provide a shifted pointer to the function. I don't think `Dataset` itself needs to support subranges.

Reply:
> in this particular case, I would actually even go as far as to make this UB (which just so happens to work as expected) if we feel like explicitly adding this support is going too far

Note that we can get all elements in the get range which is from `zero` to `total_elements`. If we allow `indptr` to have a sub-range, this is confusing.